### PR TITLE
refactor(mojaloop/#2883): move repo code out of command module and into shared lib 

### PR DESCRIPTION
**Pull request by kleyow**
refactor(mojaloop/#2883): move repo code out of command module and into shared lib - [https://github.com/mojaloop/project/issues/2883](https://github.com/mojaloop/project/issues/2883)

- Move repo code out of command module and into shared lib
- Add redis config for domain module
- Resolve pre-commit issue

### CODE REVIEW

File: modules/outbound-command-event-handler/src/application/handler.ts [hunk 1]
```diff
@@ -38,11 +39,10 @@ import {
     ProcessSDKOutboundBulkPartyInfoRequestMessage,
     ProcessPartyInfoCallbackMessage,
     ProcessSDKOutboundBulkPartyInfoRequestCompleteMessage,
+    IBulkTransactionEntityRepo,
+    ICommandEventHandlerOptions,
```
Comment (mdebarros) [inline modules/outbound-command-event-handler/src/application/handler.ts:43]: Why was `ICommandEventHandlerOptions` moved?

File: modules/private-shared-lib/src/types/command_event_handler_options.ts [hunk 2]
```diff
@@ -23,8 +23,7 @@
  ******/
 
 'use strict';
-import { IDomainEventProducer } from '@mojaloop/sdk-scheme-adapter-private-shared-lib';
-import { IBulkTransactionEntityRepo }  from './bulk_transaction_entity_repo';
+import { IBulkTransactionEntityRepo, IDomainEventProducer } from './';
 
 export type ICommandEventHandlerOptions = {
```
Comment (mdebarros) [inline modules/private-shared-lib/src/types/command_event_handler_options.ts:28]: Same comment as above, why was this moved?

File: modules/private-shared-lib/tsconfig.json [hunk 3]
```diff
@@ -1,10 +1,18 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-	"rootDir": "./src",
-	"outDir": "./dist"
+    "rootDir": "./src",
+    "outDir": "./dist",
+    "paths": {
```
Comment (mdebarros) [inline modules/private-shared-lib/tsconfig.json:6]: Is `baseUrl` not required here?

```suggestion
    "baseUrl": ".",
    "paths": {
```

File: package.json [hunk 4]
```diff
@@ -21,12 +21,14 @@
     "modules/*"
   ],
   "scripts": {
+    "postinstall": "yarn husky install",
```
Comment (mdebarros) [inline package.json:24]: Thank you!

**Review by mdebarros — CHANGES_REQUESTED** [review]

File: modules/outbound-command-event-handler/src/application/handler.ts (hunk 1 above)
Comment (kleyow) [reply modules/outbound-command-event-handler/src/application/handler.ts:43]: I forgot to move it back along with the aggregate code, sorry . Fixed.

File: modules/private-shared-lib/tsconfig.json (hunk 3 above)
Comment (kleyow) [reply modules/private-shared-lib/tsconfig.json:6]: Thanks, I forgot to add it back.

File: modules/outbound-command-event-handler/src/domain/bulk_transaction_agg/handlers/process_sdk_outbound_bulk_party_info_request.ts [hunk 5]
```diff
@@ -26,14 +26,15 @@
...
 import { ILogger } from '@mojaloop/logging-bc-public-types-lib';
 import {
+    BulkTransactionInternalState,
     CommandEvent,
+    IndividualTransferInternalState,
     ProcessSDKOutboundBulkPartyInfoRequestCmdEvt,
     PartyInfoRequestedDmEvt,
 } from '@mojaloop/sdk-scheme-adapter-private-shared-lib';
-import { BulkTransactionAgg } from '..';
+import { BulkTransactionAgg } from '@module-domain';
```
Comment (mdebarros) [inline modules/outbound-command-event-handler/src/domain/bulk_transaction_agg/handlers/process_sdk_outbound_bulk_party_info_request.ts:35]: Does this work fine?

Don't we have the self-reference issue that we had to resolve the other day?

File: modules/private-shared-lib/test/unit/infra/api_server.todo.ts [hunk 6]
```diff
@@ -1,11 +1,14 @@
...
+import { DefaultLogger } from '@mojaloop/logging-bc-client-lib';
 import axios from 'axios'
 import { ApiServer, TApiServerOptions } from '../../../src/index'
-import {ILogger} from '@mojaloop/logging-bc-public-types-lib';
+import { ILogger } from '@mojaloop/logging-bc-public-types-lib';
 
 const logger: ILogger = new DefaultLogger('bc', 'appName', 'appVersion');
 
-describe('Api Server', () => {
+describe.skip('Api Server', () => {
```
Comment (mdebarros) [inline modules/private-shared-lib/test/unit/infra/api_server.todo.ts:11]: Please add a comment as to why its being skipped.

File: modules/private-shared-lib/test/unit/infra/kafka_message_publisher.todo.ts [hunk 7]
```diff
@@ -1,7 +1,11 @@
+// NOTE: Where are the classes supposed to come from?
+//       These tests seem to be imported from somewhere.
+
+/*
 import { KafkaMessagePublisher } from '../../../src/'
 import { IMessage, MessageTypes } from '@mojaloop/sdk-scheme-adapter-private-shared-lib'
 
-describe('Kafka Message Publisher', () => {
+describe.skip('Kafka Message Publisher', () => {
```
Comment (mdebarros) [inline modules/private-shared-lib/test/unit/infra/kafka_message_publisher.todo.ts:8]: Add comment for skipping pls.

**Review by mdebarros — CHANGES_REQUESTED** [review]
Looks good, just some minor comments for you to address.

File: modules/private-shared-lib/test/unit/infra/api_server.todo.ts (hunk 6 above)
Comment (kleyow) [reply modules/private-shared-lib/test/unit/infra/api_server.todo.ts:11]: My bad, this isn't test specific. The problem is with the imports.
I've removed the `skip`. The whole file will be commented out and that explaination is at the top of the file.

File: modules/outbound-command-event-handler/src/domain/bulk_transaction_agg/handlers/process_sdk_outbound_bulk_party_info_request.ts (hunk 5 above)
Comment (kleyow) [reply modules/outbound-command-event-handler/src/domain/bulk_transaction_agg/handlers/process_sdk_outbound_bulk_party_info_request.ts:35]: Good catch. Yes it does seem to work fine but I've changed it here and throughout just incase.

**Review by mdebarros — APPROVED** [review]
+1
